source/docs/software/examples-tutorials/drivesim-tutorial/simulation-instance.rst

```diff
@@ -0,0 +1,73 @@
+Step 1: Creating Simulated Instances of Hardware
+================================================
+The WPILib simulation framework contains several ``XXXSim`` classes, where ``XXX`` represents physical hardware such as encoders or gyroscopes. These simulation classes can be used to set positions and velocities (for encoders) and angles (for gyroscopes) from a model of your drivetrain.
+
+Simulating Encoders
+-------------------
+The ``EncoderSim`` class allows users to set encoder positions and velocities on a given ``Encoder`` object. When running on real hardware, the ``Encoder`` class interacts with real sensors to count revolutions (and convert them to distance units automatically if configured to do so); however, in simulation there are no such measurements to make. The ``EncoderSim`` class can accept these simulated readings from a model of your drivetrain.
+
+.. note:: It is not possible to simulate to simulate encoders that are directly connected to CAN motor controllers using WPILib classes. For more information about your specific motor controller, please read your vendor's documentation.
```

Comment:
This language is not right

Reply:
Any suggestions for what it should look like?

Reply:
```suggestion
.. note:: It is not possible to simulate encoders that are directly connected to CAN motor controllers using WPILib classes. For more information about your specific motor controller, please read your vendor's documentation.
```

Reply:
Oh, didn't catch the double "to simulate". :palm: